app/src/lib/git/branch.ts

```diff
@@ -78,7 +78,13 @@ export async function deleteBranch(
     await deleteLocalBranch(repository, branch.name)
   }
 
-  const remoteName = branch.remote
+  let remoteName = branch.remote
+
+  if (branch.type === BranchType.Remote) {
+    remoteName = branch.name
+      .replace(branch.nameWithoutRemote, '')
+      .replace('/', '')
+  }
```

Comment:
Feels a bit odd having this logic here in addition to the `let remoteName = branch.remote` 🤔  I think I'd do one of these:
1. Adding a comment explaining why we need this (i.e. explain `branch.remote` doesn't give us the remote of remote branches).
2. Renaming the property `remote` of `Branch` to something more explicit, like `upstreamRemote`, and then encapsulate this logic in a new property, maybe something like `remotePrefix` (to be consistent with https://github.com/desktop/desktop/blob/development/app/src/lib/remove-remote-prefix.ts/#L9) and maybe a new `remote` property in `Branch` that chooses between `remotePrefix` and `upstreamRemote` depending on their availability and/or type of branch?

I think I'd go with (2). At least the part of renaming `remote` to `upstreamRemote`. I think just calling it `remote` is misleading because you would expect to get _something_ from remote branches too 😢

Reply:
I definitely agree; however I am hesitant to change what remote is as it could be fairly impactful. I did add a upstreamRemote and add documentation to remote explaining what happens when you use it for a remote branch.

Reply:
Yeah, I understand how it's a big change 😕 

The reason why I suggested to rename the current `remote` (in the class and wherever is used) to `upstreamRemote` is because it describes its behavior more accurately: it doesn't always return the remote of the branch, it only does when the branch is local and has an upstream.

The new `upstreamRemote` you just added is returning the upstream remote for local branches, which is perfect 👍, but also the branch remote for remote branches (what a 👅 🌪️  🤣), which technically don't have an upstream.

That's why I think their names it would be semantically more accurate if the names of those properties were swapped. What do you think?